src/core/namespace.h

```diff
@@ -150,8 +150,8 @@ int setup_tmp_dirs(
                 char **tmp_dir,
                 char **var_tmp_dir);
 
-int setup_netns(const int netns_storage_socket[static 2]);
-int open_netns_path(const int netns_storage_socket[static 2], const char *path);
+int setup_ns(const int ns_storage_socket[static 2], unsigned long nstype);
+int open_ns_path(const int ns_storage_socket[static 2], const char *path, unsigned long nstype);
```

Comment:
Maybe these two functions should be called something like "setup_shareable_ns"? We have other helpers that setup namespaces, and the purpose of these specific ones is to have sockets that can be stored and shared